src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,43 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        if (Bulldozer.class.equals(type)) {
+            List<? extends Machine> machines = new BulldozerProducer().get();
+            return new ArrayList<>(machines);
+        } else if (Excavator.class.equals(type)) {
+            List<? extends Machine> machines = new ExcavatorProducer().get();
+            return new ArrayList<>(machines);
+        } else if (Truck.class.equals(type)) {
+            List<? extends Machine> machines = new TruckProducer().get();
+            return new ArrayList<>(machines);
+        }
```

Comment:
producers should be `private final` class-level fields
```suggestion
        if (Bulldozer.class.equals(type)) {
            return new ArrayList<>(bulldozerProducer.get());
        } else if (Excavator.class.equals(type)) {
            return new ArrayList<>(excavatorProducer.get());
        } else if (Truck.class.equals(type)) {
            return new ArrayList<>(truckProducer.get());
        }
```